cli/cmd/sync2.go

```diff
@@ -0,0 +1,148 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/cloudquery/plugin-sdk/clients"
+	"github.com/cloudquery/plugin-sdk/specs"
+	"github.com/rs/zerolog/log"
+	"github.com/schollz/progressbar/v3"
+	"golang.org/x/sync/errgroup"
+)
+
+
+func syncConnectionV2(ctx context.Context, cqDir string,sourceClient *clients.SourceClient, sourceSpec specs.Source, destinationsSpecs []specs.Destination, uid string) error {
+	var err error
+	destinationNames := make([]string, len(destinationsSpecs))
+	for i := range destinationsSpecs {
+		destinationNames[i] = destinationsSpecs[i].Name
+	}
+	syncTime := time.Now().UTC()
+
+	log.Info().Str("source", sourceSpec.Name).Strs("destinations", destinationNames).Time("sync_time", syncTime).Msg("Start sync")
+	defer log.Info().Str("source", sourceSpec.Name).Strs("destinations", destinationNames).Time("sync_time", syncTime).Msg("End sync")
+
+	destClients := make([]*clients.DestinationClient, len(sourceSpec.Destinations))
+	destSubscriptions := make([]chan []byte, len(sourceSpec.Destinations))
+	for i := range destSubscriptions {
+		destSubscriptions[i] = make(chan []byte)
+	}
+	defer func() {
+		for _, destClient := range destClients {
+			if destClient != nil {
+				if err := destClient.Terminate(); err != nil {
+					log.Error().Err(err).Msg("Failed to terminate destination client")
+					fmt.Println("failed to terminate destination client: ", err)
+				}
+			}
+		}
+	}()
+	tables, err := sourceClient.GetTables(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to get tables for source %s: %w", sourceSpec.Name, err)
+	}
+	for i, destinationSpec := range destinationsSpecs {
+		destClients[i], err = clients.NewDestinationClient(ctx, destinationSpec.Registry, destinationSpec.Path, destinationSpec.Version,
+			clients.WithDestinationLogger(log.Logger),
+			clients.WithDestinationDirectory(cqDir),
+		)
+		if err != nil {
+			return fmt.Errorf("failed to create destination plugin client for %s: %w", destinationSpec.Name, err)
+		}
+		if err := destClients[i].Initialize(ctx, destinationSpec); err != nil {
+			return fmt.Errorf("failed to initialize destination plugin client for %s: %w", destinationSpec.Name, err)
+		}
+
+		if err := destClients[i].Migrate(ctx, tables); err != nil {
```

Comment:
@yevgenypats Maybe now is a good opportunity to change this so that we only migrate the tables specified in the config? This would also be nice for csv, since only the files for tables specified in the config will get created.

Reply:
Agree. I think they are quite a few improvements that we can/need to do. for example also to spawn only one process per unique destination. but I suggest doing this in follow-up smaller PRs. should be also easier to review and easier/faster for me to develop